client/my-sites/checkout/checkout/secure-payment-form.jsx

```diff
@@ -259,11 +251,19 @@ export class SecurePaymentForm extends Component {
 		}
 	}
 
-	finishIfLastStep( cart, selectedSite, step ) {
+	async finishIfLastStep( cart, selectedSite, step ) {
```

Comment:
This is just a personal taste thing, but I'd make the `handleTransactionStep` and `componentDidUpdate` methods `async` too, and add appropriate `await` calls. It won't make any difference to the behaviour, it might just be less surprising for our future-selves, otherwise there's no hint that something async is going to happen.

Reply:
Done in [077f738](https://github.com/Automattic/wp-calypso/pull/37356/commits/077f738510acab687b63f51000604dfca2b2c999)

Reply:
Can we also add the `await` keyword here https://github.com/Automattic/wp-calypso/blob/077f738510acab687b63f51000604dfca2b2c999/client/my-sites/checkout/checkout/secure-payment-form.jsx#L237 and here https://github.com/Automattic/wp-calypso/blob/077f738510acab687b63f51000604dfca2b2c999/client/my-sites/checkout/checkout/secure-payment-form.jsx#L83

That way the execution of the `componentDidUpdate` method really will pause until the site setting has finished saving.